t/issue/1401.t

```diff
@@ -0,0 +1,86 @@
+
+use Test::More tests => 6;
+
+use strict;
+use warnings;
+use 5.010;
+use autodie qw(:all);
+use English qw($OSNAME -no_match_vars);
+use File::Temp;
+
+use Rex::Commands::Run;
+
+$::QUIET = 1;
+
+my $win = $OSNAME =~ m/^MSWin/mxsi;
+my $quote = q{"};
+my $space = q{ };
+my $empty = q{};
+
+sub command {
+  my ( $path, $exe, @parm ) = @_;
+  return qq{$path/$exe } . join $empty, @parm;
+}
+
+sub relative {
+  return qq{./$_[0]};
+}
+
+my $path    = File::Temp->newdir( 'path with spaces XXXX', DIR => q{./} );
+my $tail    = 'hello';
+my $parm    = 'Rex is wonderfull';
+my $testcmd = '/bin/echo';
```

Comment:
Would this exist on all platform where Rex might be run or what Rex might be asked to manage? Could it be that this is located elsewhere on special/ancient/etc. systems (like `/usr/bin/echo`)?

Can we avoid hardcoding the path here, e.g. by discovering the path to `echo` with `can_run('echo')`?

Also, perhaps `$path_to_echo` is a more readable variable name here.

Reply:
You are right its that most of the times I cannot assume or install things on machine and I involuntarily code the most things from the base instead of search the available functions or objects in the system